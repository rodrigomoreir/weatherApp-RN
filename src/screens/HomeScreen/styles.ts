import styled from 'styled-components/native';

export const StyledContainer = styled.View`
  flex: 1;
  align-items: center;
  justify-content: center;
  background-color: ${({ theme }) => theme.colors.background} ;
`;

export const StyledFlatlist = styled.FlatList`
  flex: 1;
  width: 100%;
  padding: 10px 15px;
`;

export const StyledSearchContainer = styled.View`
  width: 100%;
  padding: 0 15px 10px;
`
