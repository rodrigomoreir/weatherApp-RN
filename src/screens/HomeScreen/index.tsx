import React, { useEffect } from 'react';
import { Header } from '../../components/Header';
import { WeatherCard } from '../../components/WeatherCard';

import api from '../../services/api';
import axios from 'axios';

import { getCitys } from '../../hook/getCitys';

import { StyledContainer, StyledFlatlist } from './styles';

interface Props {
  id: number;
  name: string;
  hour: string;
  weather: string;
  temperature: number;
  temperatureMin: number;
  temperatureMax: number;
}

export const HomeScreen = () => {
  var data: Props[] = []
  //   {
  //     id: londonData.data.id,
  //     name: londonData.data.name,
  //     hour: londonData.data.dt,
  //     weather: londonData.data.weather[0].description,
  //     temperature: londonData.data.main.temp,
  //     temperatureMin: londonData.data.main.temp_min,
  //     temperatureMax: londonData.data.main.temp_max
  //   },
  //   {
  //     id: 2,
  //     title: 'Brasil'
  //   }
  // ]

  // const getWeather = async () => {
  //   try {
  //     const londonData = await api.get('weather?q=London,uk&APPID=3e3afcb7e55ba8f7f9116213fd0ea726');
  //     const london = {
  //       id: londonData.data.id,
  //       name: londonData.data.name,
  //       hour: londonData.data.dt,
  //       weather: londonData.data.weather[0].description,
  //       temperature: londonData.data.main.temp,
  //       temperatureMin: londonData.data.main.temp_min,
  //       temperatureMax: londonData.data.main.temp_max
  //     }

  //     data = [london]

  //     console.log('response', londonData.data.main)
  //     console.log('name', london)
  //   } catch (err) {
  //     console.log('err', err)
  //   }
  // }

  useEffect(() => {
    getCitys('London,uk')
  }, [])

  return (
    <>
      <Header />
      <StyledContainer>
        <StyledFlatlist
          data={data}
          keyExtractor={(item) => String(item.id)}
          renderItem={({ item }) => (
            <WeatherCard
              city={item.name}
              hour={'hour'}
              weather={'weather'}
              temperature={'temp'}
              tempMaxAndMin={'temp'}
            />
          )}
        />
      </StyledContainer>
    </>
  )
};
