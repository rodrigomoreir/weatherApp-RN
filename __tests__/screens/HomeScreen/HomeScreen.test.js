import React from 'react'
import HomeScreen from '../../../src/screens/HomeScreen'

import {
  renderWithTheme,
  matchSnapshotWithTheme,
} from '../../../src/helpers/testing'

describe('Home Screen', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<HomeScreen />)
  })

  test('renders correctly', () => {
    renderWithTheme(<HomeScreen />)
  })
})
