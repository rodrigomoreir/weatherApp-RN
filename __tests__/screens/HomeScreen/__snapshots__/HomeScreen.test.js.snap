// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen matches snapshot 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 25,
            "fontWeight": "bold",
            "marginBottom": 10,
          },
        ]
      }
    >
      Weather
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 0,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#F5F8FA",
              "borderColor": "#808080",
              "borderRadius": 15,
              "borderWidth": 0.5,
              "flexDirection": "row",
              "height": 40,
            },
          ]
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Search a city"
          placeholderTextColor="#969CB2"
          style={
            Array [
              Object {
                "color": "#000000",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingLeft": 20,
              },
            ]
          }
          testID="text-input"
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginRight": 20,
              "opacity": 1,
            }
          }
          testID="pressable-icon"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/icons/icSearch.png",
              }
            }
            style={
              Array [
                Object {
                  "tintColor": "#808080",
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <RCTScrollView
      ListEmptyComponent={[Function]}
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
            "width": "100%",
          },
        ]
      }
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 20,
                  "fontStyle": "italic",
                },
              ]
            }
          >
            Sorry, nothing found
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>,
]
`;
